src/components/common/Image/Image.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from "react";
+import clsx from "clsx";
+import defaultPlaceholderUrl from "url:/assets/placeholder.png";
+import { sleep } from "~utils/promises/sleep";
+import { useTheme } from "~components/embed/contexts/ThemeContext";
+
+import styles from "./Image.module.scss";
+
+const fakeDelay = process.env.NODE_ENV === "development" ? 2500 : 0;
+
+type BorderRadiusVariant = "none" | "rounded" | "circular";
+
+interface ImageProps {
+  src: string;
+  srcDark?: string;
+  width: number;
+  height: number;
+  title?: string;
+  alt?: string;
+  className?: string;
+  fullWidth?: boolean;
+  border?: boolean;
+  borderRadius?: BorderRadiusVariant;
+  backgroundColor?: string;
+  placeholderURL?: string;
+  pointer?: [number, number];
+  style?: React.CSSProperties;
+}
+
+export default function Image({
+  src,
+  srcDark,
+  width,
+  height,
+  title,
+  alt = "",
+  className,
+  fullWidth,
+  border,
+  borderRadius = "none",
+  backgroundColor = "transparent",
+  placeholderURL = defaultPlaceholderUrl,
+  pointer: pointerProp,
+  style,
+}: ImageProps) {
+  const theme = useTheme();
+  const displayTheme = theme.mode;
+
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  const pictureClassName = clsx(className, styles.root, {
+    [styles.fullWidth]: fullWidth,
+    [styles.hasBorder]: border,
+    [styles.isRounded]: borderRadius === "rounded",
+    [styles.isCircular]: borderRadius === "circular",
+    [styles.isLoaded]: isLoaded,
+  });
+
+  const pictureStyle = {
+    "--imgWidth": `${width}px`,
+    "--imgHeight": fullWidth ? "auto" : `${height}px`,
+    "--imgAspectRatio": fullWidth ? width / height : "auto",
+    "--imgBackgroundColor": backgroundColor,
+    "--imgPlaceholderBackgroundURL": placeholderURL && placeholderURL !== "none" ? `url(${placeholderURL})` : "none",
+    ...style,
+  };
+
+  const imgClassName = clsx(styles.img, {
+    [styles.isLoaded]: isLoaded,
+  });
+
+  const handleImageLoaded = async () => {
+    if (fakeDelay) await sleep(fakeDelay + Math.random() * fakeDelay * 2);
+
+    setIsLoaded(true);
+  };
+
+  const [devPointer, setDevPointer] = useState<[number, number] | null>(null);
+
+  const pointer = devPointer || pointerProp;
+
+  const movePointer =
+    process.env.NODE_ENV === "development"
+      ? (e: React.MouseEvent<HTMLSpanElement>) => {
+          const { offsetX, offsetY } = e.nativeEvent;
+          const { offsetWidth, offsetHeight } = e.currentTarget;
+
+          const nextPointer = (
+            fullWidth ? [(100 * offsetX) / offsetWidth, (100 * offsetY) / offsetHeight] : [offsetX, offsetY]
+          ) satisfies [number, number];
+
+          console.log("pointer =", nextPointer);
+
+          setDevPointer(nextPointer);
+        }
+      : undefined;
+
+  return (
+    <picture className={pictureClassName} style={pictureStyle}>
+      {srcDark ? (
+        <source srcSet={displayTheme === "light" ? src : srcDark} media="(prefers-color-scheme: dark)" />
```

Comment:
This can easily be extended later to support more modern formats and different sizes/resolutions. Not doing any of that now to avoid making this PR longer and because we'd probably want to look into automating that if possible.